### PR TITLE
Various YouTube Integration improvements and fixes

### PR DESCRIPTION
**Pull request by derrod**
### Description
These changes are intended to improve how YouTube API errors are exposed to the user and fix a few issues found during debugging/testing.

### Motivation and Context
Several issues such as live streaming not being enabled on a channel or a Google account without YouTube channel being used during the OAuth process are currently not properly explained to the user.

The API to retrieve existing broadcasts was used poorly, leading to potentially hundreds of requests made only to throw their results away since *every single broadcast* on a channel was fetched, rather than only those we care about (live, scheduled). This PR also adds an indicator to the list informing the user that data is still being fetched from the API.

Additionally resuming a stream that was interrupted (e.g. due to OBS crashing) was not possible, and the code for using an already attached `liveStream` didn't work.

### How Has This Been Tested?
Tested with YouTube channel without livestreaming and with Google account without YouTube channel.

Tested various live streams in different configurations (scheduled, autostart on/off, etc.)

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by derrod** [comment]
This is how the selection screen looks including the YT Studio broadcast and an interrupted stream that can be resumed:
screenshot of new event selection screen

**Comment by mufunyo** [comment]
> screenshot of new event selection screen

"scheduled for" starts with a lowercase letter while the other descriptions start with a capital.

**Comment by derrod** [comment]
Yeah that's from the original translations, I changed it to upper-case now to match the rest.

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -1217,15 +1218,21 @@ YouTube.Actions.Error.NoBroadcastCreated="Broadcast creation error '%1'.<br/>A d
...
 YouTube.Actions.Error.YouTubeApi="YouTube API Error. Please see the log file for more information."
 
+YouTube.Actions.EventsLoading="Loading list of events..."
 YouTube.Actions.EventCreated.Title="Event Created"
 YouTube.Actions.EventCreated.Text="Event successfully created."
 
 YouTube.Actions.ChooseEvent="Choose an Event"
 YouTube.Actions.Stream="Stream"
-YouTube.Actions.Stream.ScheduledFor="scheduled for"
+YouTube.Actions.Stream.ScheduledFor="Scheduled for"
```
Comment (Vainock) [inline UI/data/locale/en-US.ini:1227]: Could you add a variable for the date? -> "Scheduled for %1"
Some language have a different word order, making this string currently hard to translate.

**Review by Vainock — CHANGES_REQUESTED** [review]

File: UI/data/locale/en-US.ini (hunk 1 above)
Comment (derrod) [reply UI/data/locale/en-US.ini:1227]: Yes I can.

**Comment by derrod** [comment]
I've learned an important lesson about rebasing here.

Unfortunately that lesson is "please for the love of god never try to do *that* again".

**Pull request merged**